Added power control to APU, changing naming style

molleer opened the pull request:
Test ROM 01 and 06 from dmg_sound now pass

molleer committed:
Updated boot_lameboy_big

molleer committed:
Added Setup_Sound to all boot rom versions

molleer committed:
Started implementing status control register

molleer committed:
Removed left right functionality because it does not work propperly

molleer committed:
Implemented more real world accurate read and write function

molleer committed:
Got test rom dmg_sound/01-register working

molleer committed:
Merge with dev

molleer committed:
Passing dmg_sound/06

molleer committed:
Fixed breaking changes from previous commit

molleer committed:
Fixed variable and function names

molleer committed:
Removed commented code and todos

molleer committed:
Added back optimization

molleer committed:
Removed unintended changes

Riddarvid commented on roms/asm/boot_g_tile.asm:
```diff
@@ -37,6 +39,28 @@ Start:
+		ld	[unk_FF11], a	; Sound	wave duty 50%
+		ld	a, $F3 ; 'ó'
+		ld	[unk_FF12], a	; Init Envelope
+		ld	[unk_FF25], a	; Set enabled soudn channels
```
soudn

Riddarvid commented on roms/asm/boot_lameboy.asm:
```diff
@@ -46,6 +48,28 @@ Animation:
+		ld	[unk_FF11], a	; Sound	wave duty 50%
+		ld	a, $F3 ; 'ó'
+		ld	[unk_FF12], a	; Init Envelope
+		ld	[unk_FF25], a	; Set enabled soudn channels
```
soudn

Riddarvid commented on roms/asm/boot_lameboy_big.asm:
```diff
@@ -101,6 +103,27 @@ LoadTiles:
+		ld	[unk_FF11], a	; Sound	wave duty 50%
+		ld	a, $F3 ; 'ó'
+		ld	[unk_FF12], a	; Init Envelope
+		ld	[unk_FF25], a	; Set enabled soudn channels
```
soudn

Riddarvid reviewed (changes requested):
Ser bra ut, bara en felskrivning som har kopierats!

AlgotAxelzon reviewed (changes requested):
Testsen fungerar inte för mig, men det verkar ligga en fix i `removing-labhelper` så fixar väl det med den PR:en. 
Verkar som att du har lagt till kod i .asm filerna men inte tagit med de kompilerade filerna.
Är väldigt tight med space i boot_lameboy_big.gb (den vi använder) men jag har fixat det i nästkommande commit. 
Istället för att initiera wave-ramet från boot ROM:et så görs det nu i APU::reset().

AlgotAxelzon committed:
Initiate sound registers in boot ROMs

Optimized "LameBoy" logo tiles and added sound register initialization.
Initialization of wave RAM is done in APU::reset()

AlgotAxelzon committed:
Travis

AlgotAxelzon committed:
Fix comments
